docs/discovery/index.md

```diff
@@ -613,6 +620,71 @@ func (a *Agent) validateAPI(apiID, stageName string) bool {
 
 Returning true from the validator will indicate that the *ConsumerInstance* is still valid. The Amplify Agents SDK will not remove the resource. Returning false will indicate to the Amplify Agents SDK that the resource should be removed, thereby keeping the resources and the APIs in sync.
 
+## Registering handler for API Server events
+The Agent SDK maintains cache of API server resources for its internal processing. These caches are populated by fetching the API server resource using API call which are polled at regular interval. With the support for gRPC based watch on API server resource events, the Agent SDK does not require to make the API call periodically instead uses the gRPC based watch service to receive events on API server resources. The Agent SDK creates a gRPC connection with the service and subscribes to the service based on WatchTopic resource in API server. The Agent SDK initialization create the WatchTopic resource which defines the filters based on the type of agent and type of API server resources it need for its internal processing.
```

Comment:
**a** cache

Reply:
using API **calls** which

Reply:
the Agent SDK does not require ~~to make the API call periodically instead uses~~ **making periodic API calls. Instead, it uses the gRPC based watch**

Reply:
based on **a** WatchTopic resource in API Server.

Reply:
it **needs** for internall processing